runtime/src/accounts_db.rs

```diff
@@ -2586,26 +2583,28 @@ impl AccountsDb {
         let mut alive = 0;
         let mut dead = 0;
         iter.for_each(|(pubkey, stored_account)| {
-            let lookup = self.accounts_index.get_account_read_entry(pubkey);
-            if let Some(locked_entry) = lookup {
-                let is_alive = locked_entry.slot_list().iter().any(|(_slot, i)| {
-                    i.store_id == stored_account.store_id
-                        && i.offset == stored_account.account.offset
+            self.accounts_index
+                .get_mut_function(pubkey, |locked_entry| {
```

Comment:
Note: One place we hold the write lock

Reply:
metrics on mnb indicate ratios of ~14k alive, 8k dead. So, perhaps it makes sense to pre-flight the list and identify the pubkeys which are dead and then only get the write lock for those.